### PR TITLE
Update UpgradeWizard.aspx to require a superuser to choose whether to keep Telerik or to remove Telerik

### LINKED ISSUE #4946: Update `UpgradeWizard.aspx`, require an superuser to choose whether to keep Telerik and remain insecure or to remove Telerik, IF, no dependencies are found.  If the user selects to keep Telerik, the remainder of this process remains unchanged.

### PR DESCRIPTION
**Pull request by daguiler**
Closes #4946

## Summary
This pull request changes a lot of files, but each commit is self-contained and this should help reviewing it.

### Relocated TelerikUtils for reuse 429c1721bd55c1e2e7f867df8e3527f77cf95925
This commit moves the TelerikUtils class (and the interface) from the Dnn.PersonaBar.Extensions project to the core library. This was necessary in order to reuse the Telerik detection code in the Upgrade Wizard.
### Fixed TelerikUtils locking files in bin folder 1740b7cfeccf348525783bf7751683752c520f20
The Telerik detection logic basically scans all assemblies in the bin folder, loads them to the app domain, and checks their dependencies. The problem was that all assemblies would stay loaded afterwards and the files in the bin folder would remain locked. This didn't create issues for #5077, but it certainly created issues for the upgrade process. The solution I found was to use a temporary AppDomain to do the check, and then discard it.
### TelerikUtils skip system files d10657aabc6021f9189d175191b9ec23c208e436
This commit basically optimizes the Telerik check by skipping system files.
### Added Telerik check to Upgrade Wizard [`86410e8 + 1bb6d1f`](https://github.com/daguiler/Dnn.Platform/compare/d10657aabc6021f9189d175191b9ec23c208e436...daguiler:1bb6d1ff45f823e9194622fc3e2df1a8771c4e1e)
And finally this is the commit that includes all the UI changes to the Upgrade Wizard.

### CODE REVIEW

File: DNN Platform/Website/Install/UpgradeWizard.aspx [hunk 1]
```diff
@@ -57,15 +57,23 @@
...
                          <div class="dnnWizardStep">
-                            <span class="dnnWizardStepNumber">2</span>
+                            <span class="dnnWizardStepNumber">3</span>
                             <span class="dnnWizardStepTitle"><%= LocalizeString("Upgrade")%></span>
                             <span class="dnnWizardStepArrow"></span>
                         </div>
                         </a>
                      </li>
-                    <li><a href="#upgradeViewWebsite">
+                    <li><a href="#upgradeInstallation">
```
Comment (bdukes) [inline DNN Platform/Website/Install/UpgradeWizard.aspx:76]: Was this supposed to be changed?

**Review by bdukes — APPROVED** [review]

**Review by valadas — APPROVED** [review]
Excellent, thank you very much

File: DNN Platform/Website/Install/UpgradeWizard.aspx (hunk 1 above)
Comment (daguiler) [reply DNN Platform/Website/Install/UpgradeWizard.aspx:76]: Yes, that was on purpose. The thing is that the `upgradeViewWebsite` link was pointing to an empty `<div>`, unnecessarily. And there was some Javascript code to handle this case. With this change, when you click on this link it just stays in the installation progress panel.

**Review by donker — APPROVED** [review]
Great. Thanks.

**Pull request merged**